hrm-service/src/case/case-data-access.ts

```diff
@@ -198,32 +199,43 @@ export const deleteById = async (id, accountSid) => {
 
 export const update = async (
   id,
-  body: Partial<CaseRecord>,
+  caseRecordUpdates: Partial<NewCaseRecord>,
   accountSid,
-  workerSid,
+  toAuditRecord = c => c,
 ): Promise<CaseRecord> => {
-  delete body.accountSid;
   const result = await db.tx(async transaction => {
-    const caseRecordUpdates: Partial<NewCaseRecord> = {
-      ...body,
-      twilioWorkerId: workerSid,
-      updatedAt: new Date().toISOString(),
-    };
-
-    const updateById = updateByIdSql(caseRecordUpdates);
-    const [original, updated] = await transaction.multi<CaseRecord>(updateById, {
-      accountSid,
-      caseId: id,
-    });
-
+    const caseUpdateSqlStatements = [selectSingleCaseByIdSql('Cases')];
+    if (caseRecordUpdates.info) {
+      const allSections: CaseSectionRecord[] = caseRecordUpdates.caseSections ?? [];
+      if (allSections.length) {
+        caseUpdateSqlStatements.push(caseSectionUpsertSql(allSections));
+      }
+      // Map case sections into a list of ids grouped by category, which allows a more concise DELETE SQL statement to be generated
+      const caseSectionIdsByType = allSections.reduce((idsBySectionType, caseSection) => {
+        idsBySectionType[caseSection.sectionType] = idsBySectionType[caseSection.sectionType] ?? [];
+        idsBySectionType[caseSection.sectionType].push(caseSection.sectionId);
+        return idsBySectionType;
+      }, <Record<string, string[]>>{});
+      caseUpdateSqlStatements.push(deleteMissingCaseSectionsSql(caseSectionIdsByType));
```

Comment:
> DELETE any case sections present in the existing data with a section type / section ID combination that is NOT present in the blob specified in the PUT.

Ha! You told explained here. Followup question: can't this result in accidentally deleting things if the frontend sends a partial update of a case?

Reply:
I believe in the previous logic updates to the `info` field was always assumed to be a replacement version rather than a merge - partial updating only applied to the top level columns on the `Cases` table

To replicate this logic faithfully - if `households` is missing from an updated `info` blob supplied in a `PUT`, the assumption is the case now has no household entries, and if there were any households on the case before, you want to delete them all